src/shell/Session.ts

```diff
@@ -22,7 +22,7 @@ export class Session extends EmitterWithUniqueID {
     history = History;
     historicalPresentDirectoriesStack = new OrderedSet<string>();
 
-    constructor(private application: ApplicationComponent, private _dimensions: Dimensions) {
+    constructor(private application: ApplicationComponent, private _dimensions: Dimensions, public position: Positions) {
```

Comment:
@shockone actually you need to separate `TabComponent` from the `TabHeaderComponent` that pushed me add position here.. because it was only one place to add it safety